src/goImport.ts

```diff
@@ -9,31 +9,95 @@ import vscode = require('vscode');
 import cp = require('child_process');
 import { getBinPath } from './goPath';
 import { parseFilePrelude } from './util';
-import { promptForMissingTool } from './goInstallTools';
 import { documentSymbols } from './goOutline';
+import { promptForMissingTool, isVendorSupported } from './goInstallTools';
+import path = require('path');
 
 export function listPackages(excludeImportedPkgs: boolean = false): Thenable<string[]> {
 	let importsPromise = excludeImportedPkgs && vscode.window.activeTextEditor ? getImports(vscode.window.activeTextEditor.document.fileName) : Promise.resolve([]);
-	let pkgsPromise = new Promise<string[]>((resolve, reject) => {
+	let vendorSupportPromise = isVendorSupported();
+	let goPkgsPromise = new Promise<string[]>((resolve, reject) => {
 		cp.execFile(getBinPath('gopkgs'), [], (err, stdout, stderr) => {
 			if (err && (<any>err).code === 'ENOENT') {
 				promptForMissingTool('gopkgs');
 				return reject();
 			}
 			let lines = stdout.toString().split('\n');
-			let sortedlines = lines.sort().slice(1); // Drop the empty entry from the final '\n'
-			return resolve(sortedlines);
+			return resolve(lines);
 		});
 	});
 
-	return Promise.all<string[]>([importsPromise, pkgsPromise]).then(values => {
-		let imports = values[0];
-		let pkgs = values[1];
-		if (imports.length === 0) {
-			return pkgs;
-		}
-		return pkgs.filter(element => {
-			return imports.indexOf(element) === -1;
+	return vendorSupportPromise.then((vendorSupport: boolean) => {
+		return Promise.all<string[]>([goPkgsPromise, importsPromise]).then(values => {
+			let pkgs = values[0];
+			let importedPkgs = values [1];
+
+			if (!vendorSupport || !vscode.window.activeTextEditor) {
+				if (importedPkgs.length > 0) {
+					pkgs = pkgs.filter(element => {
+						return importedPkgs.indexOf(element) === -1;
+					});
+				}
+				return pkgs.sort().slice(1);
```

Comment:
the stdout from gopkgs is a `\n` separated pkg paths. So when you do a split, you end up with a trailing empty string. That's what this originally was, and I just moved the code around while refactoring. Will fix it nevertheless.